WcaOnRails/app/webpacker/lib/hooks/useRequest.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { fetchWithAuthenticityToken } from '../requests/fetchWithAuthenticityToken';
+
+export default function useRequest(successTrigger = null, errorTrigger = null) {
+  const [loading, setLoading] = React.useState(false);
+  const [data, setData] = React.useState(null);
+  const [error, setError] = React.useState(null);
```

Comment:
Since these three are very dependent (when one updates, usually most/all of the others also update in sync) a reducer would make a lot of sense. Then instead of
```
      setLoading(true);
      setData(null);
      setError(null);
```
you could just `dispatch({ action: 'reset' } )`, etc.

Reply:
Ignoring this as this file is removed now. :-)